background.js

```diff
@@ -8,56 +8,102 @@ var cdnDomains = [
     'cdnjs.cloudflare.com',
     'code.jquery.com',
     'aui-cdn.atlassian.com',
+    'npmcdn.com',
+    'fonts.googleapis.com'
+    // add your other CDN here!
 ];
 
 /***** END CONFIG *****/
 
-
-///// Storage and options
-
-var filterDomains = cdnDomains.map(function (domain) {
-    return '*://' + domain + '/*';
-});
 var webRequestFilter = {
-    urls: filterDomains,
+    urls: cdnDomains.map(function (domain) {
+        return '*://' + domain + '/*';
+    }),
     types: ['stylesheet', 'script', 'image', 'object', 'xmlhttprequest']
 };
-var beforeRequestOptions = ['blocking'];
-var cacheIsActive = false;
 
+// see https://developer.chrome.com/apps/offline_storage
+var storage = chrome.storage.local;
 
-///// Helper methods
+// internal copy of the chrome storage
+// TODO: investigate performance & memory consumption
+var cache = {}
 
-/* Source: https://developer.mozilla.org/en-US/docs/Web/API/WindowBase64/Base64_encoding_and_decoding#The_.22Unicode_Problem.22 */
+/* see https://developer.mozilla.org/en-US/docs/Web/API/WindowBase64/Base64_encoding_and_decoding#The_.22Unicode_Problem.22 */
 function b64EncodeUnicode(str) {
-    return btoa(encodeURIComponent(str).replace(/%([0-9A-F]{2})/g, function(match, p1) {
+    return btoa(encodeURIComponent(str).replace(/%([0-9A-F]{2})/g, function (match, p1) {
         return String.fromCharCode('0x' + p1);
     }));
 }
 
+function amountOfItemsInCache() {
+    return '' + Object.keys(cache).length;
+};
 
 ///// Caching and filter implemention
+function addListener() {
+
+    // load storage first, see: http://stackoverflow.com/a/18579032
+    storage.get(null, function (_cache) {
+        cache = _cache;
```

Comment:
Good feedback.

1. cache contents should not change. that's the idea behind a CDN! :smile:
2. i'm also not sure about the exact behavior. on the one hand I don't want to waste even more memory, on the other hand I want to have it permanently stored. i was expecting that the data should survive a browser open/close, as long as you do not deactivate the extension.  but to be honest, I would need to double-check that! :-)